libraries/standard/json/src/json.c

```diff
@@ -0,0 +1,1014 @@
+#include <assert.h>
+#include <ctype.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "json.h"
+
+typedef enum bool
+{
+    true = 1, false = 0
+} bool;
+
+/**
+ * @brief Advance buffer index beyond whitespace.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ */
+static void skipSpace( const char * buf,
+                       size_t * start,
+                       size_t max )
+{
+    size_t i = *start;
+
+    for( ; i < max; i++ )
+    {
+        if( isspace( buf[ i ] ) == 0U )
+        {
+            break;
+        }
+    }
+
+    *start = i;
+}
+
+/**
+ * @brief Count the leading 1s in a byte.
+ *
+ * The high-order 1 bits of the first byte in a UTF-8 encoding
+ * indicate the number of additional bytes to follow.
+ *
+ * @return the count
+ */
+static size_t countHighBits( uint8_t c )
+{
+    uint8_t n = c;
+    size_t i = 0U;
+
+    while( ( n & 0x80U ) != 0U )
+    {
+        i++;
+        n <<= 1;
+    }
+
+    return i;
+}
+
+/**
+ * @brief Advance buffer index beyond a UTF-8 code point.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ *
+ * @return true if a valid code point was present;
+ * false otherwise.
+ */
+static bool skipUTF8MultiByte( const char * buf,
```

Comment:
this is a bit horrendous, but you may have to deal with the security considerations section of https://tools.ietf.org/html/rfc3629#page-10. I probably wrote the bug exploited in the "widespread virus attacking Web servers in 2001". I'd prefer not be responsible for a second version of the same attack

Reply:
Challenge accepted!  Tell me if you think I've got it covered.